sdk/subscribe_block.go

```diff
@@ -24,6 +25,96 @@ const (
 	pathCosignature        = "cosignature"
 )
 
+// Closes the subscription channel.
+func closeChannel(s *subscribe) {
```

Comment:
I think you need to pass as a parameter channel immediately.
Or make method closeChannel as inside method *subcsribe